### ISSUE TITLE
Pyglet-based integrations end in infinite recursion on closing window under macOS

### ISSUE THREAD

**Issue by swistakm** [swistakm/pyimgui#159]
Happens with following dependency versions:
```
pyglet==1.4.3
cocos2d==0.6.7
```

Traceback:
```
Traceback (most recent call last):
  File "_ctypes/callbacks.c", line 234, in 'calling callback function'
  File "/Users/swistakm/dev/pyimgui/venv/lib/python3.6/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 1121, in objc_method
    result = f(py_self, *args)
  File "/Users/swistakm/dev/pyimgui/venv/lib/python3.6/site-packages/pyglet/window/cocoa/pyglet_window.py", line 65, in nextEventMatchingMask_untilDate_inMode_dequeue_
    if self.inLiveResize():
  File "/Users/swistakm/dev/pyimgui/venv/lib/python3.6/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 785, in __call__
    return self.method(self.objc_id, *args)
  File "/Users/swistakm/dev/pyimgui/venv/lib/python3.6/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 755, in __call__
    result = f(objc_id, self.selector, *args)
ctypes.ArgumentError: ("argument 1: <class 'RecursionError'>: maximum recursion depth exceeded while calling a Python object", "selector = b'inLiveResize'", "argtypes =[<class 'ctypes.c_void_p'>, <class 'ctypes.c_void_p'>]", "encoding = b'c16@0:8'")
Traceback (most recent call last):
  File "_ctypes/callbacks.c", line 234, in 'calling callback function'
  File "/Users/swistakm/dev/pyimgui/venv/lib/python3.6/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 1121, in objc_method
    result = f(py_self, *args)
  File "/Users/swistakm/dev/pyimgui/venv/lib/python3.6/site-packages/pyglet/window/cocoa/pyglet_window.py", line 65, in nextEventMatchingMask_untilDate_inMode_dequeue_
    if self.inLiveResize():
  File "/Users/swistakm/dev/pyimgui/venv/lib/python3.6/site-packages/pyglet/libs/darwin/cocoapy/runtime.py", line 785, in __call__
```

**Comment by swistakm** [comment]
OK, this seems to be error of pyglet (https://github.com/pyglet/pyglet/issues/5) itself fixed in version [1.5.6](https://github.com/pyglet/pyglet/releases/tag/v1.5.6).

I'm going to resolve this issue by providing minimal requirement of `pyglet>=1.5.6` on darwin platform for `imgui[pyglet]`and `imgui[cocos2d]`

**Comment by swistakm** [comment]
Fixed in 1.2.0
